### DISCUSSION TITLE
Starlight is missing a pattern to translate localized folders, subfolder names

### DISCUSSION THREAD

**Discussion by DamianChybowski95Plus** [withastro/starlight#2058] [Feature Requests]
### What version of `starlight` are you using?

0.24.4

### What version of `astro` are you using?

4.11.1

### What package manager are you using?

npm

### What operating system are you using?

Windows x64

### What browser are you using?

Edge

### Describe the Bug

It's more of a design bug, or a missing part of documentation, than an emergency. Astro / starlight doesn't have a pattern for localizing a subfolders name. For example docfx implements it with .override files, that can be used to translate subfolder names for other use cases than routing. I would suggest simply to add a naming convention to folders, so that they could be localized easily. 

![t](https://github.com/withastro/starlight/assets/146984890/0aee8622-d3c6-4626-8a8f-4e01501bdb7a)

For example: FolderName [[ llocalizedName ]], would mean that FolderName is used for routing, and localized name can be used for everything else.

### Link to Minimal Reproducible Example

design or documentation bug, so it doesn't require an example

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by delucis** [comment]
Thanks for sharing @DamianChybowski95Plus — moved this to a feature request as, like you said, it’s not exactly a bug, more a design limitation/new feature.
